### PR TITLE
use Wireshark to check traces

### PR DESCRIPTION
**Pull request by marten-seemann**
DRAFT: do not merge yet.

### CODE REVIEW

File: interop/trace.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  FROM_CLIENT = 1
+  FROM_SERVER = 2
+
+class TraceAnalyzer:
+  _filename = ""
+
+  def __init__(self, filename: str):
+    self._filename = filename
+
+  def _get_dirction_filter(self, d: Direction) -> str:
```
Comment (janaiyengar) [inline interop/trace.py]:
```suggestion
  def _get_direction_filter(self, d: Direction) -> str:
```

File: interop/trace.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    if d == Direction.FROM_CLIENT:
+      return "ip.src==192.168.0.100 && "
+    elif d == Direction.FROM_SERVER:
+      return "ip.src==192.168.100.100 && "
+    else:
+      return ""
+
+  def get_retry(self, direction: Direction = Direction.ALL) -> pyshark.FileCapture:
+    f = self._get_dirction_filter(direction) + "quic.long.packet_type==Retry"
```
Comment (janaiyengar) [inline interop/trace.py]:
```suggestion
    f = self._get_direction_filter(direction) + "quic.long.packet_type==Retry"
```

File: interop/trace.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    return pyshark.FileCapture(self._filename, display_filter=f)
+
+  def get_initial(self, direction: Direction = Direction.ALL) -> pyshark.FileCapture:
+    """ Get all Initial packets.
+    Note that this might return coalesced packets. Filter by:
+    packet.quic.long_packet_type == "0"
+    """
+    f = self._get_dirction_filter(direction) + "quic.long.packet_type==Initial"
```
Comment (janaiyengar) [inline interop/trace.py]:
```suggestion
    f = self._get_direction_filter(direction) + "quic.long.packet_type==Initial"
```

File: interop/trace.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    return pyshark.FileCapture(self._filename, display_filter=f)
+
+  def get_handshake(self, direction: Direction = Direction.ALL) -> pyshark.FileCapture:
+    """ Get all Initial packets.
+    Note that this might return coalesced packets. Filter by:
+    packet.quic.long_packet_type == "2"
+    """
+    f = self._get_dirction_filter(direction) + "quic.long.packet_type==Handshake"
```
Comment (janaiyengar) [inline interop/trace.py]:
```suggestion
    f = self._get_direction_filter(direction) + "quic.long.packet_type==Handshake"
```

File: interop/trace.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
+from enum import Enum
+
+import pyshark
+
+cap = pyshark.FileCapture("../logs/sim/trace_node_left.pcap", display_filter="ip.dst==192.168.0.100 && quic")
```
Comment (janaiyengar) [inline interop/trace.py]: Why do you need this line?

File: interop/trace.py [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+      return ""
+
+  def get_retry(self, direction: Direction = Direction.ALL) -> pyshark.FileCapture:
+    f = self._get_dirction_filter(direction) + "quic.long.packet_type==Retry"
+    return pyshark.FileCapture(self._filename, display_filter=f)
+
+  def get_initial(self, direction: Direction = Direction.ALL) -> pyshark.FileCapture:
+    """ Get all Initial packets.
+    Note that this might return coalesced packets. Filter by:
+    packet.quic.long_packet_type == "0"
```
Comment (janaiyengar) [inline interop/trace.py]: I don't understand this comment (and below). If this returns coalesced packets, how does a caller separate the packets out?

**Comment by janaiyengar** [comment]
This doesn't work with quicly... I wonder if it has to do with wireshark's handling of coalesced packets.

File: interop/trace.py (hunk 5 above)
Comment (marten-seemann) [reply interop/trace.py]: Fixed.

File: interop/trace.py (hunk 6 above)
Comment (marten-seemann) [reply interop/trace.py]: `long_packet_type == 0` makes sure that this is actually an Initial packet.

**Comment by marten-seemann** [comment]
Let's get this merged and add verification for the other test cases later.

**Pull request merged**